addons/udes_sale_stock/models/stock_move.py

```diff
@@ -27,15 +27,17 @@ def _action_cancel(self):
         def not_cancelled_filter(m):
             return m.state not in ["cancel"] and m.location_dest_id == location_customers
 
-        if not self.env.context.get("disable_sale_cancel", False):
-            lines_to_cancel = (
-                self.filtered(lambda m: m.location_dest_id == location_customers)
-                .mapped("sale_line_id")
-                .filtered(lambda s: len(s.move_ids.filtered(not_cancelled_filter)) == 0)
-            )
-            if lines_to_cancel:
-                lines_to_cancel.action_cancel()
+        delivery_cancelled_moves = self.filtered(lambda m: m.location_dest_id == location_customers)
 
+        if not self.env.context.get("disable_sale_cancel", False) and delivery_cancelled_moves:
+            lines_to_cancel = delivery_cancelled_moves.sale_line_id.filtered(lambda s: len(s.move_ids.filtered(not_cancelled_filter)) == 0)
```

Comment:
This is the only place that we use `not_cancelled_filter`, so perhaps we should replace it with its inverse to simplify the code?  It's always harder to think about `not`s.